### DISCUSSION TITLE
Ignite 8.8.0 - can't launch ios simulator

### DISCUSSION THREAD

**Discussion by Eric2048** [infinitered/ignite#2429] [Q&A] [answered]
Tried a first-time user install of Ignite, following [https://github.com/infinitered/ignite/blob/master/docs/Guide.md](https://github.com/infinitered/ignite/blob/master/docs/Guide.md)  and can't get past the "Welcome to Metro" step.
Steps:
`npx ignite-cli@latest new AppTest`
I then answered the questions, package manager: npm, allowed "install deps" etc.
-> Creating AppTest using Ignite 8.8.0
`npm start`
This started Metro, which shows its "... i - run on iOS ..." (etc) messages.

**Q1: The next step in the instructions is 'npm run ios' -- but Metro is running. Should I ignore those instructions and press 'i', or start a new terminal to run 'npm run ios' ?**

**Q2: I then tried pressing 'i' in Metro:**
```
info Opening the app on iOS...
info Found Xcode workspace "AppTest.xcworkspace"
info Found booted Eric’s iPhone (2)
info Run CLI with --verbose flag for more details.
```
...but the simulator window does not open. Note: previous to this I had run the react native setup instructions ([](https://reactnative.dev/docs/environment-setup?guide=native)) and was able to from their launch the simulator and side-load the app into "Eric's iPhone", so perhaps there is some Xcode setting which is directing Metro to attempt to send this new Ignite app to my phone?

In case this helps:
```
npx ignite-cli doctor
System
  platform           darwin
  arch               x64
  cpu                16 cores     Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
  directory          jiseki       /Users/eric/jiseki

JavaScript (and globally-installed packages)
  node               16.14.2      /usr/local/bin/node
  npm                8.5.0        /usr/local/bin/npm
    bot-builder-ui   0.1.0
    builder-ui       0.1.0
    corepack         0.10.0
    ios-deploy       1.12.2
    n                8.0.2
    npm              8.5.0
  yarn               -            not installed
  pnpm               -            not installed
  expo               -            not installed

Ignite
  ignite-cli         8.8.0        /Users/eric/.npm/_npx/e31027f3785124a8/node_modules/.bin/ignite
  ignite src         build        /Users/eric/.npm/_npx/e31027f3785124a8/node_modules/ignite-cli/build

Android
  java               1.8.0_321    /usr/bin/java
  android home       -            undefined

iOS
  xcode              14.3
  cocoapods          1.12.1       /Users/eric/.rvm/rubies/ruby-2.7.6/bin/pod

Tools
  git                git version 2.39.2 (Apple Git-143)   /usr/bin/git
```

Thanks in advance for any help.

**Comment by frankcalise** [comment]
Hi Eric, is your phone connected to your computer? If that's the case it may default to that.

However, you should be able to specify the simulator name by appending `--simulator="iPhone 14 Pro"` for example to `yarn ios`.

Also you could start via Expo, which generally is a quicker way to get up and running via `yarn expo:ios` or `yarn expo:start`

**Reply by Eric2048** [reply]
The first time I tried npm start, my iPhone was not connected. I later connected it and saw the same error message. After seeing your reply I disconnected my phone, tried `npm start --simulator="iPhone 14 Pro"`, and (after two tries) successfully launched the app -- but in iPhone SE.
It seems to ignore the simulator arg -- I even tried the simulator's identifier instead of its name, that did not work. The output is:
```
info No booted devices or simulators found. Launching first available simulator...
info Launching iPhone SE (3rd generation) (iOS 16.4)
```
Also tried `npm run ios --simulator="iPhone 14 Pro"`, same result.
I also opened this new Ignite-created project in Xcode for the first time, thinking I had skipped that step in the Ignite setup: File->Open ~/AppTest/ios, and in the XCode editor I can see a Device Name at the top, with a menu to select a Device, but selecting "iPhone 14 Pro" does not have any effect on the which simulator gets started.
Also checked Xcode WIndow->Devices and Simulators, and for "iPhone 14 Pro" confirmed that "Show run destination" is set to "Automatic" (not "Never").
So thanks for getting me over the initial hurdle -- any recommendations on how to fix the device selection?

**Reply by frankcalise** [reply]
Do you actually have a simulator named "iPhone 14 Pro"? That was just an example, but you can see the names of them if you open the Simulator File menu > Open Simulator > they are listed there.

You shouldn't have to bother opening Xcode at all, I typically don't

**Reply by Eric2048** [reply, marked as answer]
Yes in the Simulator "iPhone 14 Pro" is listed in that menu, and I can launch it. With both the iPhone SE sim and iPhone 14 Pro sim running, `npm start` will always select the SE. With `npm start --simulator="iPhone 14 Pro"` it will first launch the App in SE:
```
info Found booted iPhone SE (3rd generation), iPhone 14 Pro
info Launching iPhone SE (3rd generation)
```
and then it also launches it in the 14 Pro:
```
success Successfully launched the app on the simulator
info Launching iPhone 14 Pro

```
With just the 14 Pro sim running, it launches the app in that.
After more experimentation: it seems that `npm start` and `npm run ios` are both ignoring the --simulator argument, and to select a specific phone I need to start that simulator ahead of time. Just surprised that this runs against the Ignite docs and that others are not running into this. I'll mark this as the answer in case anyone else runs into this, thanks.

**Reply by jamonholmgren** [reply]
Note you can also run `open -a Simulator.app` and it'll open ... something.
